src/components/Feed/Feed.jsx

```diff
@@ -0,0 +1,204 @@
+        }
+
+        const data = await response.json();
+        return data.filter((comment) => !commentIdSet.has(comment.id));
+      } catch (error) {
+        console.error("Error fetching comment data:", error);
+      }
+    };
+
+    setCommentData([]);
+    commentIdSet.clear();
+
+    const fetchCommentsPromises = postData.map((post) => fetchPostComments(post.id));
+    
+    let isMounted = true;
+
+    Promise.all(fetchCommentsPromises).then((commentsArrays) => {
+      if (isMounted) {
+        const uniqueComments = [].concat(...commentsArrays).filter(comment => {
+          if (commentIdSet.has(comment.id)) {
+            return false;
+          }
+          commentIdSet.add(comment.id);
+          return true;
+        });
+        setCommentData(uniqueComments);
+      }
+    });
+
+    return () => {
+      isMounted = false;
+    };
+  }, [postData]);
+
+  const handlePostSubmit = async () => {
+    try {
+      const response = await fetch('https://boolean-api-server.fly.dev/irlydo/post', {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify({
+          title: 'New post',
+          content: newPostContent,
+          contactId: 1,
+        }),
+      });
+
+      console.log('POST Request Response:', response);
+      const newPost = await response.json();
+
+      setPostData((prevData) => [newPost, ...prevData]);
+
+      setNewPostContent('');
+    } catch (error) {
+      console.error('Error submitting post:', error);
+    }
+  };
+
+  const handleCommentSubmit = async (postId) => {
+    try {
+      if (!postId) {
+        console.error('postId is required for submitting a comment');
+        return;
+      }
+
+      const response = await fetch(`https://boolean-api-server.fly.dev/irlydo/post/${postId}/comment`, {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify({
+          content: newCommentContent,
+          contactId: 1, 
+          postId: postId
+        }),
+      });
+
+      const newComment = await response.json();
+
+      setCommentData((prevData) => [...prevData, newComment]);
+
+      setNewCommentContent('');
+    } catch (error) {
+      console.error('Error submitting comment:', error);
+    }
+  };
+
+  return (
+    <ul className="feed-ul">
+      <input
+        type='text'
+        placeholder="What's on your mind?"
+        value={newPostContent}
+        onChange={(e) => setNewPostContent(e.target.value)}
+      ></input>
+      <button type="button" className='submit-button' onClick={handlePostSubmit}>
+      <i className="fa-solid fa-paper-plane"></i>
+      </button>
+      {postData.map((post) => {
+        const contact = contactData.find((contact) => contact.id === post.contactId);
+        if (!contact) {
+          console.error(`Contact not found for post with id ${post.id}`);
```

Comment:
Good use of clear console logs in your error handling to identify what might be causing problems throughout your code.